src/adera/Machines/Rocket.h

```diff
@@ -80,9 +81,22 @@ class SysMachineRocket :
 class MachineRocket : public osp::active::Machine
 {
     friend SysMachineRocket;
+    struct ResourceInput
+    {
+        osp::DependRes<ShipResourceType> m_type;
+        float m_massRateFraction;
+        osp::active::ActiveEnt m_sourceEnt;
+    };
+    using fuel_list_t = std::vector<ResourceInput>;
```

Comment:
This arrangement kind of sucks, and will be changed in the next PR or two when the connections are automated/generalized